mapper/container_ports.go

```diff
@@ -98,13 +98,11 @@ func (n ContainerPorts) ToInternal(data map[string]interface{}) error {
 					logrus.Warnf("Failed to encode port: %v", err)
 					return obj
 				}
-				if strings.EqualFold(convert.ToString(mapped["kind"]), "HostPort") {
-					if _, ok := mapped["sourcePort"]; ok {
-						mapped["hostPort"] = mapped["sourcePort"]
-					}
+				if !strings.EqualFold(convert.ToString(mapped["kind"]), "HostPort") {
+					// delete the source port so it doesn't get converted to the host port by default mapper
+					delete(mapped, "sourcePort")
```

Comment:
Is there a possibility of losing the information for nodePort/LB since we will not be saving the port info to the etcd? In case there is a delay in the controllers launching  the corresponding nodeport/LB services?

Reply:
no @prachidamle. It gets preserved in "port" annotation.